mountpoint-s3/src/main.rs

```diff
@@ -371,16 +372,25 @@ fn mount(args: CliArgs) -> anyhow::Result<FuseSession> {
         .use_accelerate(args.transfer_acceleration)
         .use_dual_stack(args.dual_stack);
 
-    let throughput_target_gbps =
-        args.maximum_throughput_gbps
-            .map(|t| t as f64)
-            .unwrap_or_else(|| match calculate_network_throughput() {
-                Ok(throughput) => throughput,
-                Err(e) => {
-                    tracing::warn!("failed to detect network throughput: {:?}", e);
-                    DEFAULT_TARGET_THROUGHPUT
-                }
-            });
+    let region = args.region.or_else(|| {
+        let region = env_region();
+        if let Some(region) = &region {
+            tracing::debug!("using AWS_REGION: {region}");
+        };
+        region
+    });
```

Comment:
push this logic into `get_region`? check the env var first if `provided_region` is None

Reply:
Slight complication that the retry logic needs to know whether the user provided a region. But I'll think how to do that.